src/saltext/vmware/states/vmc_sddc.py

```diff
@@ -0,0 +1,384 @@
+            (Optional) The ID of the customer connected account to work with.
+
+        'customer_subnet_ids': Array of String
+            (Optional) List of subnet Ids
+
+    deployment_type: String
+        (Optional) Denotes if request is for a SingleAZ or a MultiAZ SDDC. Default is SingleAZ.
+        Possible values are: SingleAZ, MultiAZ
+
+    host_instance_type: String
+        (Optional) The instance type for the ESX hosts in the primary cluster of the SDDC.
+        Possible values are: i3.metal, r5.metal, i3en.metal
+
+    msft_license_config :
+        (Optional) Indicates the desired licensing support, if any, of Microsoft software.
+        It can be specified in the below format
+
+        .. code::
+
+            msft_license_config
+                - academic_license: False
+                - mssql_licensing: "DISABLED"
+                - windows_licensing: "DISABLED"
+
+             where,
+             academic_license - Flag to identify if it is Academic Standard or Commercial Standard License.
+             mssql_licensing - The status MSSQL licensing for this SDDC’s clusters.
+                                Possible values are: DISABLED, CUSTOMER_SUPPLIED, ENABLED
+             windows_licensing - The status of Windows licensing for this SDDC’s clusters. Can be enabled, disabled, or customer’s.
+                                Possible values are: DISABLED, CUSTOMER_SUPPLIED, ENABLED
+
+        Please refer the `VMC Doc about msft_license_config <https://developer.vmware.com/apis/vmc/v1.1/data-structures/MsftLicensingConfig/>`_
+
+    sddc_id: String As UUID
+        (Optional)If provided, will be assigned as SDDC id of the provisioned SDDC.
+
+    sddc_template_id : String As UUID
+        (Optional) If provided, configuration from the template will applied to the provisioned SDDC.
+
+    sddc_type: String
+        (Optional)Denotes the SDDC type, if the value is null or empty, the type is considered as default.
+
+    size: String
+        (Optional) The size of the vCenter and NSX appliances. “large” sddcSize corresponds to a ‘large’ vCenter
+        appliance and ‘large’ NSX appliance. ‘medium’ sddcSize corresponds to ‘medium’ vCenter appliance and
+        ‘medium’ NSX appliance. Value defaults to ‘medium’.
+        Possible values are: nsx_small, medium, large, nsx_large
+
+    skip_creating_vxlan : Boolean
+        (Optional) skip creating vxlan for compute gateway for SDDC provisioning
+
+    sso_domain : String
+        (Optional) The SSO domain name to use for vSphere users. If not specified, vmc.local will be used.
+
+    storage_capacity:  Integer As Int64
+        (Optional) The storage capacity value to be requested for the SDDC primary cluster, in GiBs. If provided,
+        instead of using the direct-attached storage, a capacity value amount of seperable storage will be used.
+
+    vpc_cidr
+        (Optional) AWS VPC IP range. Only prefix of 16 or 20 is currently supported.
```

Comment:
Do you mean a _suffix_ of `/16` or `/20`?  Otherwise, I assume you are saying `16.0.0.0/8` or `20.0.0.0/8`.

Reply:
it expects value in this format 0.0.0.0/16, so as you said it should be suffix

But here we have written wrapper on VMC API, and their document says prefix

so i have added an example but did not change the wording as it may confuse user if they refer the API doc

Reply:
According to https://www.ripe.net/about-us/press-centre/understanding-ip-addressing and https://www.rfc-editor.org/rfc/rfc4632#page-5 in particular:

>    For example, the legacy "Class B" network 172.16.0.0, with an implied
>   network mask of 255.255.0.0, is defined as the prefix 172.16.0.0/16,

So weirdly enough, I think that "prefix" is an accurate term. Also https://developer.vmware.com/apis/vmc/v1.1/data-structures/SddcConfig/ for the api docs